doc/building.md

```diff
@@ -0,0 +1,221 @@
+      Under no circumstances should NuGet packages be committed to source control.
+- The solution is built
+    - `msbuild` (on Windows systems) and `xbuild` (on Unix-like systems) is executed on the solution file (`CKAN.sln`).
+      This causes each project file (`.csproj`) to be built producing an assembly in `_build/out`. Each project file
+      has the following lines which control its output directory. Any new projects should include these lines:
+      ```xml
+      <OutputPath>..\_build\out\$(AssemblyName)\$(Configuration)\bin\</OutputPath>
+      <IntermediateOutputPath>..\_build\out\$(AssemblyName)\$(Configuration)\obj\</IntermediateOutputPath>
+      ```
+- Exeutables are repacked
+    - ILRepack is executed to merge the CKAN executable and all its dependencies and the NetKAN executable and all its
+      dependencies into singular executable files. This allows the CKAN and NetKAN clients to be distributed as single
+      files without complex install procedures. These are the "final" output of the build process and are stored in:
+      `_build/repack/$CONFIGURATION`.
+- Unit tests are executed
+    - The NUnit unit tests in `CKAN.Tests.dll` are executed
+- Smoke tets are executed
+    - These are simple tests designed to make sure there isn't anything grossly wrong with the build. All they do is
+      execute the repacked `ckan.exe` and `netkan.exe` and make sure their version output matches the expected output.
+
+## Travis
+
+Travis is the continuous integration (CI) environment that is used to automatically build CKAN and NetKAN. Travis's
+configuration is controlled by `.travis.yml` in the top-level directory. Upon every push Travis will execute the cross
+product of mono versions and build configurations. As there are currently three specified mono versions (`4.6.2`,
+`4.2.4`, and `4.0.5`) and two specified build configurations (`Debug`, `Release`), Travis will execute *six* builds per
+push. Travis's operation is pretty simple:
+
+- A bunch of packages are installed to make the build work.
+- Some commands are executed to simulate a graphical environment for testing.
+- The solution is built using `./build --configuration=$BUILD_CONFIGURATION`
+- The solution is tested using `./build test+only --configuration=$BUILD_CONFIGURATION --exclude=FlakyNetwork`
+    - The `--exclude=FlakyNetwork` argument is used to not execute tests that rely upon a network connection and could
+      thus behave nondeterministically (frustrating behavior for a CI system).
+- The built executables (`ckan.exe` and `netkan.exe`) are uploaded to Amazon S3 if the following conditions are met:
+    - All previous steps were successful
+    - Its on the `master` branch
+    - The build configuraton is `Release`
+    - The mono version used is the latest (currently `4.6.2`)
+- A GitHub release is generated if the following conditions are met:
+    - All previous steps were successful
+    - The commit is tagged
```

Comment:
The [current method](https://github.com/KSP-CKAN/CKAN/wiki/Releasing-a-new-CKAN-client) is to go to the releases page and create a new release there. So we still do that, but now the version number reported by the client will come from CHANGELOG.md. Ok, I get it. At some point I must grokify how that actually happens.